liborbum/src/Controller/Ee/Vpu/Vu/Interpreter/CVuInterpreter_FLOAT.cpp

```diff
@@ -8,52 +10,132 @@
 
 void CVuInterpreter::ABS(VuUnit_Base* unit, const VuInstruction inst)
 {
-    // TODO: Implement.
-#if defined(BUILD_DEBUG)
-    BOOST_LOG(Core::get_logger()) << boost::format("(%s, %d) ABS: Not implemented.") % __FILENAME__ % __LINE__;
-#else
-    throw std::runtime_error("ABS: Not implemented.");
-#endif
+    SizedQwordRegister& reg_source = unit->vf[inst.fs()];
+    SizedQwordRegister& reg_dest = unit->vf[inst.ft()];
+
+    for (auto field : VuVectorField::VECTOR_FIELDS)
+    {
+        if (inst.test_dest_field(field))
+        {
+            // std::abs uses compiler intrinsics internally, and hence it's
+            // sometimes faster than bit ops ...
+            // const float val = reg_source.read_float(field);
+            // const float result = to_ps2_float(std::abs(val), flags);
+
+            // ... but it might screw the PS2 floats out, so we do bit ops by
+            // ANDing the value with 0x7FFFFFFF, setting the first bit to 0 (positive).
+            const uword raw = reg_source.read_uword(field) & 0x7FFFFFFF;
+            const float result = *reinterpret_cast<const float*>(&raw);
+
+            reg_dest.write_float(field, result);
+        }
+    }
 }
 
 void CVuInterpreter::ADD(VuUnit_Base* unit, const VuInstruction inst)
 {
-    // TODO: Implement.
-#if defined(BUILD_DEBUG)
-    BOOST_LOG(Core::get_logger()) << boost::format("(%s, %d) ADD: Not implemented.") % __FILENAME__ % __LINE__;
-#else
-    throw std::runtime_error("ADD: Not implemented.");
-#endif
+    SizedQwordRegister& reg_source_1 = unit->vf[inst.fs()];
+    SizedQwordRegister& reg_source_2 = unit->vf[inst.ft()];
+    SizedQwordRegister& reg_dest = unit->vf[inst.fd()];
+
+    unit->status.clear_flags();
```

Comment:
i dont think this needs to be here (for any instruction) - the update_vector_field will clear/set the relevant flags, but this also touches the D and I flags, which do not change in this instruction (according to manual)

the whole clear_flags might be better off removed actually - can't remember why i added it (probably a quick fix for getting past the bios), and its useless when update_vector_field does the same thing...

instruction ok otherwise

Reply:
sorry, if it wasn't clear, i was talking about the clear_flags() call.

I know i've used it in the few instructions i put in, but taking another look at it, i believe its wrong.